File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/StatusPedidoEmbarcador.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{
    public enum StatusPedidoEmbarcador
    {
        Todos = -1,
        NaoDefinido = 0,
        Aberto = 1,
        BloqueadoPorRegra = 2,
        BloqueadoPorVerba = 3,
        AguardandoSalesForce = 4,
        Liberado = 5,
        Encerrado = 6
    }

    public static class StatusPedidoEmbarcadorHelper
    {
        public static string ObterDescricao(this StatusPedidoEmbarcador status)
        {
            switch (status)
            {
                case StatusPedidoEmbarcador.NaoDefinido: return "Não Definido";
                case StatusPedidoEmbarcador.Aberto: return "Aberto";
                case StatusPedidoEmbarcador.BloqueadoPorRegra: return "Bloqueado Por Regra";
                case StatusPedidoEmbarcador.BloqueadoPorVerba: return "Bloqueado Por Verba";
                case StatusPedidoEmbarcador.AguardandoSalesForce: return "Aguardando Sales Force";
                case StatusPedidoEmbarcador.Liberado: return "Liberado";
                case StatusPedidoEmbarcador.Encerrado: return "Encerrado";
                default: return string.Empty;
            }
        }
    }
}

File: SGT.WebAdmin/ViewsScripts/Enumeradores/StatusPedidoEmbarcador.js
var StatusPedidoEmbarcadorHelper = function () {
    this.Todos = -1;
    this.NaoDefinido = 0;
    this.Aberto = 1;
    this.BloqueadoPorRegra = 2;
    this.BloqueadoPorVerba = 3;
    this.AguardandoSalesForce = 4;
    this.Liberado = 5;
    this.Encerrado = 6;
}

StatusPedidoEmbarcadorHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "Não Definido", value: this.NaoDefinido },
            { text: "Aberto", value: this.Aberto },
            { text: "Bloqueado por Regra", value: this.BloqueadoPorRegra },
            { text: "Bloqueado por Verba", value: this.BloqueadoPorVerba },
            { text: "Aguardando Sales Force", value: this.AguardandoSalesForce },
            { text: "Liberado", value: this.Liberado },
            { text: "Encerrado", value: this.Encerrado }
        ];
    },
    obterOpcoesPesquisa: function () {
        return [{ text: "Todos", value: this.Todos }].concat(this.obterOpcoes());
    }
}

var StatusPedidoEmbarcador = new StatusPedidoEmbarcadorHelper();